modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/MessageBoardAttachmentResourceImpl.java

```diff
@@ -87,6 +107,24 @@ public MessageBoardAttachment getMessageBoardAttachment(
 		return _getMessageBoardAttachmentsPage(mbThread.getRootMessageId());
 	}
 
+	@Override
+	public MessageBoardAttachment
+			getSiteMessageBoardMessageByExternalReferenceCodeMessageBoardMessageExternalReferenceCodeMessageBoardAttachmentByExternalReferenceCode(
+				Long siteId, String messageBoardMessageExternalReferenceCode,
+				String externalReferenceCode)
+		throws Exception {
+
+		MBMessage mbMessage =
+			_mbMessageLocalService.getMBMessageByExternalReferenceCode(
+				siteId, messageBoardMessageExternalReferenceCode);
+
+		FileEntry fileEntry =
+			mbMessage.getAttachmentsFileEntryByExternalReferenceCode(
+				siteId, externalReferenceCode);
+
+		return _toMessageBoardAttachment(fileEntry);
```

Comment:
Are we sure that we are checking permissions here?? I'm seeing that we get the `mbMessage` using the `LocalService` and then get the file entry through that mbMessage.

Unknown if that method check for permissions or not but I think not. I'm asking because I'm not 100% secure (not even 60% 😅 )

Reply:
I think this endpoint should check permissions on the FileEntry and not on the thread, or maybe in both. But definitely on the FileEntry. @sergiojimcos can you verify permissions are being checked?

Reply:
I'm going to check and see if it permissions is checked

Reply:
@LuismiBarcos @4lejandrito As I could see, the endpoint should only verify the permission on the MessageBoard because an Attachment is just a FileEntry which is imported directly, not from Liferay (like Documents and Media). yo can take a view at `WikiPageAttachmentResourceImpl`

Reply:
I don't know If I understood correctly.
I check the `WikiPageAttachmentResourceImpl`, concretely, the method `deleteSiteWikiPageByExternalReferenceCodeWikiPageExternalReferenceCodeWikiPageAttachmentByExternalReferenceCode` which is the "twin" of this one.

I see this. 
```
WikiPage wikiPage =
			_wikiPageService.getLatestPageByExternalReferenceCode(
				siteId, wikiPageExternalReferenceCode);
```
We are using the external service, which checks the permissions.

If I look at this PR, I see
```
MBMessage mbMessage =
			_mbMessageLocalService.getMBMessageByExternalReferenceCode(
				siteId, messageBoardMessageExternalReferenceCode);
```

We are using the local service, which not checks the permissions.

Am I missing something?

Reply:
You are right @LuismiBarcos! What I explained before is that I should use `MBMessageService` to check the permissions but in the case of FileEntry not because an Attachment is just a FileEntry which is imported directly